table-api/src/main/java/io/deephaven/api/TableOperations.java

```diff
@@ -0,0 +1,146 @@
+    TOPS updateView(Collection<Selectable> columns);
+
+    // -------------------------------------------------------------------------------------------
+
+    TOPS update(String... columns);
+
+    TOPS update(Collection<Selectable> columns);
+
+    // -------------------------------------------------------------------------------------------
+
+    TOPS select(String... columns);
+
+    TOPS select(Collection<Selectable> columns);
+
+    // -------------------------------------------------------------------------------------------
+
+    TOPS naturalJoin(TABLE rightTable, String columnsToMatch);
+
+    TOPS naturalJoin(TABLE rightTable, String columnsToMatch, String columnsToAdd);
+
+    TOPS naturalJoin(TABLE rightTable, Collection<JoinMatch> columnsToMatch,
+        Collection<JoinAddition> columnsToAdd);
+
+    // -------------------------------------------------------------------------------------------
+
+    TOPS exactJoin(TABLE rightTable, String columnsToMatch);
+
+    TOPS exactJoin(TABLE rightTable, String columnsToMatch, String columnsToAdd);
+
+    TOPS exactJoin(TABLE rightTable, Collection<JoinMatch> columnsToMatch,
+        Collection<JoinAddition> columnsToAdd);
+
+    // -------------------------------------------------------------------------------------------
+
+    TOPS leftJoin(TABLE rightTable, String columnsToMatch);
+
+    TOPS leftJoin(TABLE rightTable, String columnsToMatch, String columnsToAdd);
+
+    TOPS leftJoin(TABLE rightTable, Collection<JoinMatch> columnsToMatch,
+        Collection<JoinAddition> columnsToAdd);
+
+    // -------------------------------------------------------------------------------------------
+
+    TOPS join(TABLE rightTable, String columnsToMatch);
+
+    TOPS join(TABLE rightTable, String columnsToMatch, String columnsToAdd);
+
+    TOPS join(TABLE rightTable, Collection<JoinMatch> columnsToMatch,
+        Collection<JoinAddition> columnsToAdd);
+
+    // -------------------------------------------------------------------------------------------
+
+    TOPS aj(TABLE rightTable, Collection<JoinMatch> columnsToMatch,
+        Collection<JoinAddition> columnsToAdd);
+
+    TOPS aj(TABLE rightTable, Collection<JoinMatch> columnsToMatch,
+        Collection<JoinAddition> columnsToAdd, AsOfJoinRule asOfJoinRule);
+
+    // -------------------------------------------------------------------------------------------
+
+    TOPS raj(TABLE rightTable, Collection<JoinMatch> columnsToMatch,
+        Collection<JoinAddition> columnsToAdd);
+
+    TOPS raj(TABLE rightTable, Collection<JoinMatch> columnsToMatch,
+        Collection<JoinAddition> columnsToAdd, RightAsOfJoinRule rightAsOfJoinRule);
+
+    // -------------------------------------------------------------------------------------------
+
+    TOPS by();
+
+    TOPS by(String... groupByColumns);
+
+    TOPS by(Collection<Selectable> groupByColumns);
+
+    TOPS by(Collection<Selectable> groupByColumns, Collection<Aggregation> aggregations);
```

Comment:
I feel bad asking for docs on so many things, but public apis w/o documentation are not very new-user-friendly.  "What is `raj()`?"  :man_shrugging: